src/collection.js

```diff
@@ -157,10 +165,10 @@ export default class Collection {
       return Promise.reject(new Error("Record is not an object."));
     }
     const newRecord = Object.assign({}, record, {
-      id:      options.synced || options.forceUUID ? record.id : uuid4(),
+      id:      options.synced || options.forceUUID ? record.id : this._idSchema.generate(),
```

Comment:
The way I understand it, getting assigned a new random UUID is the default, and the optional forceUUID parameter is so you can force your own /specific/ UUID.